### PR TITLE
`StatusMenu` refactoring

### LINKED ISSUE #7232: Refactor `StatusPopupMenu`
### Description

As stated [here in comments to PR](https://github.com/status-im/status-desktop/issues/6930#issuecomment-1209611707), `StatusPopupMenu` component needs to be refactored, some terrible things are going inside there.

### PR DESCRIPTION
**Pull request by igor-sirotin**
Closes #7232

⚠️  The code was already reviewed in:
- https://github.com/status-im/status-desktop/pull/8341
- https://github.com/status-im/status-desktop/pull/8436
- https://github.com/status-im/status-desktop/pull/8473

### What does the PR do

1. Rename menu components as follows:

| Old name | New name |
| ---------- | ----------- |
| StatusPopupMenu | StatusMenu |
| StatusMenuItem | StatusAction |
| StatusMenuItemDelegate | StatusMenuItem |

2. Fix minor bug of icon color in `StatusSearchLocationMenu`
3. Fix minor bug of accessing null properties in `StatusMenu` in `d.assetSettings`

### Affected areas

All menus across the application

### StatusQ checklist

- [ ] add documentation if necessary (new component, new feature)
- [x] update sandbox app
  - in case of new component, add new component page
  - in case of new features, add variation to existing component page
  - nice to have: add it to the demo application as well
- [x] test changes in both light and dark theme?

#### Application login

https://user-images.githubusercontent.com/25482501/204614004-e0149719-e114-4645-992d-d63ee8f5a3e1.mov

#### Application (dark)

https://user-images.githubusercontent.com/25482501/204614019-de5ee167-b1f4-4fd5-8f46-b5a93de704dd.mov

#### Application (light)

https://user-images.githubusercontent.com/25482501/204614068-a572da3a-e746-496a-bbc6-bc1dc43a3e41.mov

### CODE REVIEW

File: ui/StatusQ/src/StatusQ/Popups/StatusMenu.qml [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        implicitWidth: contentWidth
+        interactive: contentHeight > availableHeight
+        model: root.contentModel
+    }
+
+    background: Item {
+        id: backgroundItem
+        implicitWidth: 176
+
+        Rectangle {
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Popups/StatusMenu.qml:95]: You could have this Rectangle directly as the background, why the nesting? :)

File: ui/StatusQ/src/StatusQ/Popups/StatusMenuItem.qml [hunk 2]
```diff
@@ -1,25 +1,171 @@
...
+        horizontalAlignment: Text.AlignLeft
+        verticalAlignment: Text.AlignVCenter
+
+        text: root.text
+        color: !root.enabled ? Theme.palette.baseColor1
+                             : d.isStatusDangerAction ? Theme.palette.dangerColor1 : Theme.palette.directColor1
+
+        font.pixelSize: d.fontSettings.pixelSize
+        font.bold: d.fontSettings.bold
+        font.italic: d.fontSettings.italic
```
Comment (caybro) [inline ui/StatusQ/src/StatusQ/Popups/StatusMenuItem.qml:140]: What if I want to set e.g. `font.weight`?

File: ui/imports/shared/controls/chat/ProfileHeader.qml [hunk 3]
```diff
@@ -212,20 +212,20 @@ Item {
...
+    StatusMenu {
         id: imageEditMenu
 
-        StatusMenuItem {
+        StatusAction {
             text: qsTr("Upload a file")
-            icon.name: "download"
-            iconRotation: 180
+            assetSettings.name: "download"
+            assetSettings.rotation: 180
```
Comment (caybro) [inline ui/imports/shared/controls/chat/ProfileHeader.qml:221]: Don't we have a "download" icon? :thinking:

**Review by caybro — COMMENTED** [review]
Code looks good, just some minor remarks inline

Will test tomorrow

**Review by caybro — APPROVED** [review]
Tested, works fine, not a single issue spotted, well done!

**Comment by caybro** [comment]
Pls squash the commits

**Comment by caybro** [comment]
> > Pls squash the commits
> 
> @caybro, doesn't GItHub squash them when closing PR?
> 
> 

I don't think so, at least not in our setup

**Review by jrainville — APPROVED** [review]
Individual PRs have already been reviewewd

**Comment by jrainville** [comment]
> doesn't GItHub squash them when closing PR?

Nope. You can keep multiple commits if you prefer, but at least squash the ones that don't use the commit guidelines

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Pull request merged**